server/lib/report.js

```diff
@@ -1,10 +1,38 @@
 const debug = require('debug')('perf-widget:lib:areWeGettingBetter'); // eslint-disable-line no-unused-vars
-const bluebird = require('bluebird');
+const escape = require('mysql').escape;
 const query = require('./database').query;
 
-module.exports = bluebird.coroutine(function* (){
+function getHistoricalInsightsForDomain (domain){
 
+	debug('\n\n\n', domain);
+
+	//const command = `SELECT page.url, value_history.value, value_history.date FROM page JOIN value_history ON page.id = value_history.page_id WHERE page.domain=${escape(domain)};`;
+	const command = `SELECT page.url, value_history.value, value_history.date FROM page JOIN value_history ON page.id = value_history.page_id JOIN properties ON value_history.property_id=1 WHERE page.domain=${escape(domain)} GROUP BY value_history.date;`;
+
+	return query(command)
+		.then(result => {
+			debug('\n\n\n', result);
+			return {
+				data : result,
+				domain : domain,
+				str : JSON.stringify(result)
+			};
+		})
+	;
+
+}
+
+function getAllValuesFromTheLastTwentyFourHours (){
 	const command = `SELECT current_values.value, current_values.link, current_values.date, page.url, page.domain, properties.provider, properties.name FROM current_values JOIN properties ON current_values.property_id = properties.id JOIN page ON current_values.page_id = page.id WHERE date > UNIX_TIMESTAMP() - 86400 ORDER BY page.domain;`;
 
-	return query(command);
-});
+	return query(command)
+		.then(result => {
```

Comment:
The function in this thenable is the identity function -- `function (x) { return x;}`, it can safely be removed :-)